src/cookbooks/bigquery.md

```diff
@@ -0,0 +1,148 @@
+
+## From Databricks
+_Work in progress_
+
+
+## Interfaces
+BigQuery datasets and tables can be accessed by the following methods:
+
+- [re:dash](https://sql.telemetry.mozilla.org/)
+- [GCP BigQuery Console](https://console.cloud.google.com/bigquery)
+    - For advanced use cases including managing query outputs, table management. Requires GCP access to be granted by Data Operations.
+- [GCP BigQuery API Access](https://cloud.google.com/bigquery/docs/reference/rest/)
+    - For advanced use cases including automated workloads, ETL, [BigQuery Storage API](https://cloud.google.com/bigquery/docs/reference/storage/). Requires GCP access to be granted by Data Operations.
+    - Allows access to BigQuery via [`bq` command-line tool](https://cloud.google.com/bigquery/docs/bq-command-line-tool)
+- [Databricks](https://sso.mozilla.com/databricks) via BigQuery Spark Connector which uses [BigQuery Storage API](https://cloud.google.com/bigquery/docs/reference/storage/)
+
+For access to BigQuery via GCP Console and API please file a bug [here](https://bugzilla.mozilla.org/enter_bug.cgi?assigned_to=jthomas%40mozilla.com&bug_file_loc=https%3A%2F%2Fmana.mozilla.org%2Fwiki%2Fx%2FiIPeB&bug_ignored=0&bug_severity=normal&bug_status=NEW&bug_type=task&cf_fx_iteration=---&cf_fx_points=---&comment=Please%20grant%20me%20access%20to%20the%20BigQuery%20GCP%20console%20and%20API%20Access.%20I%20work%20on%20%3Cteam%3E.%0D%0A%0D%0AMy%20mozilla.com%20ldap%20login%20is%20%3Cyour%20ldap%20login%3E%40mozilla.com.&component=Operations&contenttypemethod=list&contenttypeselection=text%2Fplain&defined_groups=1&flag_type-4=X&flag_type-607=X&flag_type-800=X&flag_type-803=X&flag_type-936=X&form_name=enter_bug&maketemplate=Remember%20values%20as%20bookmarkable%20template&op_sys=Unspecified&priority=--&product=Data%20Platform%20and%20Tools&qa_contact=jthomas%40mozilla.com&rep_platform=Unspecified&short_desc=BigQuery%20GCP%20Console%20and%20API%20Access%20for%20%3Cyour%20ldap%20login%3E%40mozilla.com&target_milestone=---&version=unspecified). As part of this request we will add you to the appropriate Google Groups and provision a GCP Service Account.
+
+# Querying Tables
+
+## Projects, Datasets and Tables in BigQuery
+In GCP a [project](https://cloud.google.com/resource-manager/docs/creating-managing-projects) is a way to organize cloud resources. We use multiple
+projects to maintain our BigQuery [datasets](https://cloud.google.com/bigquery/docs/datasets-intro). BigQuery datasets are a top-level container used to organize and
+control access to tables and views.
+
+### Caveats
+
+- The date partition field (e.g. `submission_date_s3`, `submission_date`) is mostly used as a partitioning column,
+but it has changed from `YYYYMMDD` form to the more standards-friendly `YYYY-MM-DD` form.
+- Unqualified queries can become very costly very easily. We've placed restrictions on large tables from accidentally querying "all data for all time",
```

Comment:
It doesn't look like we explicitly state in this doc that BigQuery charges by data scanned, and that the data is columnar. Maybe we can add this to the intro paragraph?